src/number_theory/bernoulli.lean

```diff
@@ -257,3 +257,119 @@ begin
   { ext x, rw bernoulli_eq_bernoulli' (succ_ne_zero x ∘ succ.inj) },
   { ring },
 end
+
+
+theorem bernoulli_power_series:
+  (exp ℚ - 1) * power_series.mk (λ n, (bernoulli n / nat.factorial n)) = X :=
+ begin
+   sorry,
+ end
+
+section faulhaber
+
+lemma aux_cauchy_prod (n:ℕ):
+mk (λ (p : ℕ), bernoulli p / ↑(p.factorial)) *
+ mk (λ (q : ℕ), (coeff ℚ (q + 1)) (exp ℚ ^ n))
+ = mk (λ (p : ℕ), (finset.range (p + 1)).sum (λ (i : ℕ),
+ bernoulli i * ↑((p + 1).choose i) * ↑n ^ (p + 1 - i) / (↑(p + 1).factorial))) :=
+begin
+  ext q,
+  rw [power_series.coeff_mul],
+  simp only [coeff_mk, coeff_mk, nat.cast_mul],
+  let f: ℕ →  ℕ → ℚ := λ (a : ℕ), λ (b : ℕ),
+  bernoulli a / ↑(a.factorial) * (coeff ℚ (b + 1)) (exp ℚ ^ n),
+  rw [finset.nat.sum_antidiagonal_eq_sum_range_succ f],
+  refine finset.sum_congr _ _,
+  { refl, },
+  { intros m h,
+    simp only [f, exp_pow_eq_rescale_exp, rescale, one_div, coeff_mk, ring_hom.coe_mk, coeff_exp,
+      ring_hom.id_apply, nat.cast_mul, rat.algebra_map_rat_rat],
+    simp only [finset.mem_range] at h,
+    rw [nat.choose_eq_factorial_div_factorial (le_of_lt h)],
+    have hq: (q.succ.factorial:ℚ) ≠ 0,
```

Comment:
A general style comment: colons (not part of `:=`) should always have spaces on both sides:
```suggestion
    have hq : (q.succ.factorial : ℚ) ≠ 0,
```
This applies throughout this PR, but I won't make comments on the rest to avoid noise.

Reply:
Made those changes throughout and while at it replace a few double (and triple) space by single spaces..

Reply:
Looks like you made the changes everywhere except in this lemma!

Reply:
oops, yes. Corrected..